runbotics/runbotics-desktop/src/actions/sharepoint/SharepointActionHandler.ts

```diff
@@ -57,11 +72,26 @@ export class SharepointExcelActionHandler extends StatelessActionHandler {
         super();
     }
 
-    async open(input: SharepointOpenActionInput): Promise<SharepointOpenActionOutput> {
+    async openFileFromSite(input: SharepointOpenFromSiteActionInput): Promise<SharepointOpenActionOutput> {
+        const cloudPath = CloudPath.SITE;
         const token = await this.microsoftSession.getToken();
-        await this.microsoftService.getFileIdA41SP(token.token, input.fileName);
-        await this.microsoftService.getWorksheetId(token.token, input.worksheetName);
-        return await this.microsoftService.createSession(token.token, input.persistChanges);
+        const sharepointsiteId = await this.microsoftService.getSiteIdBySearch(token.token, input.siteName);
+        const sharepointDriveId = await this.microsoftService.getDriveId(token.token, sharepointsiteId, input.listName);
+        const sharepointFileId = await this.microsoftService.getFileIdByPath(token.token, cloudPath, input.filePath);
+        const sharepointWorksheetId = await this.microsoftService.getWorksheetId(token.token, cloudPath, input.worksheetName);
+        return await this.microsoftService.createSession(token.token, cloudPath, input.persistChanges);
+    }
+
+    async openFileFromRoot(input: SharepointOpenFromRootActionInput): Promise<SharepointOpenActionOutput> {
+        const cloudPath = CloudPath.ROOT;
+        const token = await this.microsoftSession.getToken();
+        const sharepointFileId = await this.microsoftService.getFileIdByPath(token.token, cloudPath, input.filePath);
+        const sharepointWorksheetId = await this.microsoftService.getWorksheetId(token.token, cloudPath, input.worksheetName);
```

Comment:
same here, this assignment is redundant. If these functions do something like open file, open worksheet in a file and so on then naming should be adjusted because we don't know what exactly it is doing